DirectProgramming/C++SYCL_FPGA/Tutorials/Features/IPAuthoring_interfaces_overview/handshake-control/src/vector_add.cpp

```diff
@@ -0,0 +1,105 @@
+#include <iostream>
+
+// oneAPI headers
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/experimental/fpga_kernel_properties.hpp>
+#include <sycl/ext/oneapi/annotated_arg/annotated_arg.hpp>
+#include <sycl/sycl.hpp>
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class SimpleVAdd;
+
+struct SimpleVAddKernel {
+  sycl::ext::oneapi::experimental::annotated_arg<
+      int *, decltype(sycl::ext::oneapi::experimental::properties{
+                 sycl::ext::oneapi::experimental::conduit})>
+      A_in;
```

Comment:
```suggestion
  sycl::ext::oneapi::experimental::annotated_ptr<
      int, decltype(sycl::ext::oneapi::experimental::properties{
                 sycl::ext::oneapi::experimental::conduit})>
      A_in;
```

they decided to use `annotated_ptr` for pointers and `annotated_arg` for scalars